### ISSUE TITLE
False positive: W0212: Access to a protected member in 2.5

### ISSUE THREAD

**Issue by mafrosis** [PyCQA/pylint#3607]
### Description

In version 2.5.x the following fails, and it does not in 2.4.x

```
if isinstance(f.default, dataclasses._MISSING_TYPE) and \
    isinstance(f.default_factory, dataclasses._MISSING_TYPE):  # type: ignore # pylint: disable=protected-access
```

### Steps to reproduce
1. `git clone https://github.com/mafrosis/jira-offline.git`
2. remove `pylint` version pinning in `requirements-dev.txt`
3. `docker-compose build jira-offline`
4. `make lint`

### Current behavior

```
root@90427b5f3775:/app# pylint jira_offline
************* Module jira_offline.utils.serializer
jira_offline/utils/serializer.py:173:49: W0212: Access to a protected member _MISSING_TYPE of a client class (protected-access)

-----------------------------------
Your code has been rated at 9.99/10
```

### Expected behavior

No error

### pylint --version output

```
root@6eb4b604bab8:/app# pylint --version
pylint 2.5.2
astroid 2.4.1
Python 3.7.7 (default, Apr 16 2020, 18:55:06)
[GCC 8.3.0]
```

**Comment by jacobtylerwalls** [comment]
Based on https://github.com/PyCQA/pylint/issues/3429#issuecomment-592430121 the pragma needs to be in a place containing the entire line.

One alternative if you do not want to affect the entire `if` would be to use `disable-next` to get the line with the `\`:

```python
import dataclasses
# pylint: disable-next=protected-access
if isinstance(None, dataclasses._MISSING_TYPE) and \
    isinstance(None, dataclasses._MISSING_TYPE):  # type: ignore  # pylint: disable=protected-access
    pass
```
```
python3 -m pylint ../Desktop/a.py --enable=protected-access


--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

***
So I suggest this be closed.

**Comment by Pierre-Sassoulas** [comment]
I think we won't fix issue with ``\`` for prolonging line, the cost is too high and there are multiple easy contourning solutions as @jacobtylerwalls said, including using black to format properly 

```python
if isinstance(f.default, dataclasses._MISSING_TYPE) and isinstance(
    f.default_factory, dataclasses._MISSING_TYPE
):  # type: ignore # pylint: disable=protected-access
    pass
```
